keps/sig-node/4819-log-rotate-on-disk-pressure/README.md

```diff
@@ -0,0 +1,451 @@
+              log_size=$(($log_size + 1048576))  # Increment size by 1MB
+              sleep 0.1  # Sleep to control log generation speed
+            done
+            end_time=$(date +%s)
+            echo "Log generation completed in $((end_time - start_time)) seconds"
+      restartPolicy: Never
+  backoffLimit: 4
+```
+File sizes
+```
+-rw-r----- 1 root root  24142862 Jan  1 11:41 0.log
+-rw-r--r-- 1 root root 183335398 Jan  1 11:40 0.log.20250101-113948.gz
+-rw-r--r-- 1 root root 364144934 Jan  1 11:40 0.log.20250101-114003.gz
+-rw-r--r-- 1 root root 487803789 Jan  1 11:40 0.log.20250101-114023.gz
+-rw-r--r-- 1 root root 577188544 Jan  1 11:41 0.log.20250101-114047.gz
+-rw-r----- 1 root root 730449620 Jan  1 11:41 0.log.20250101-114115
+```
+
+### Spec 2
+
+Continuously generating 10Mib with 10 sec sleep in between
+```
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: generate-huge-logs
+spec:
+  template:
+    spec:
+      containers:
+      - name: log-generator
+        image: busybox
+        command: ["/bin/sh", "-c"]
+        args:
+          - |
+            # Generate huge log entries to stdout
+            start_time=$(date +%s)
+            log_size=0
+            target_size=$((4 * 1024 * 1024 * 1024))  # 4 GB target size in bytes
+            while [ $log_size -lt $target_size ]; do
+              # Generate 1 MB of random data and write it to stdout
+              echo "Generating huge log entry at $(date) - $(dd if=/dev/urandom bs=10M count=1 2>/dev/null)"
+              log_size=$(($log_size + 1048576))  # Increment size by 1MB
+              sleep 0.1  # Sleep to control log generation speed
+            done
+            end_time=$(date +%s)
+            echo "Log generation completed in $((end_time - start_time)) seconds"
+      restartPolicy: Never
+  backoffLimit: 4
+```
+
+File sizes
+```
+-rw-r----- 1 root root 181176268 Jan  1 11:31 0.log
+-rw-r--r-- 1 root root 183336647 Jan  1 11:20 0.log.20250101-111730.gz
+-rw-r--r-- 1 root root 183323382 Jan  1 11:23 0.log.20250101-112026.gz
+-rw-r--r-- 1 root root 183327676 Jan  1 11:26 0.log.20250101-112321.gz
+-rw-r--r-- 1 root root 183336376 Jan  1 11:29 0.log.20250101-112616.gz
+-rw-r----- 1 root root 205360966 Jan  1 11:29 0.log.20250101-112911
+```
+
+
+If the pod had been generating logs in Gigabytes with minimal delay, it can cause disk pressure on kubelet host and that can affect other pods running in the same kubelets.
+
+### Goals
+
+- Rotate and Clean all container logs on kubelet Disk pressure
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
```

Comment:
we will need some details here

Reply:
brainstorming, let's evaluate some ideas:
- if the log rotation manager detects that the previous X rotations (1? 2? N?) made the rotated logs exceeds the configured max size, because containers are producing many logs continuously, should this cause kubelet to report disk pressure?
- do we want or need to review log retention also? if a rotated log is like 500 megs while it is supposed to be 100 megs (random example numbers), should _that_ cause the kubelet to report disk pressure? arguably, we are consuming more disk space than the user configured, and thus expects.

Reply:
I am not sure if these should trigger a disk pressure report. That can have a side-effect. Right ?

However, this can definitely be Turned into a warning event from the node level to indicate that something is going wrong. Would that lead to an event storm for some cases? Depends on how frequently we generate the event.

Reply:
> if the log rotation manager detects that the previous X rotations (1? 2? N?) made the rotated logs exceeds the configured max size

This problem becomes even more concerning with high value being set for `containerLogMaxFiles`.